### PR TITLE
college course drop-down menu for selecting classes, instead of text input

### PR DESCRIPTION
**Pull request by ylin250**
dokku: https://proj-ylin250.dokku-09.cs.ucsb.edu/
In this PR, I changed the school textbox input field in the course creation page into a dropdown selection instead.
This is what it looks like:
dropdown pr proj organic
Basically, the dropdown selection shown calls api/schools/all to populate the dropdown with schools created on the schools creation page.
To test the feature: make sure a school is created in the school creation page first, then navigate to the courses page and click create course and select the box in the school field to see the dropdown.

### CODE REVIEW

**Comment by ylin250** [comment]
dokku:https://proj-ylin250.dokku-09.cs.ucsb.edu/
This pr changed course creation from entering school as text into selecting a dropdown menu of schools.
After creating component for the dropdown in frontend, api calls were made to feed the information from backend into the array of school names for the dropdown to be displayed. Some tests and other files are modified to accommodate for the change.

**Review by JUstISAGE — DISMISSED** [review]
LGTM!

File: frontend/src/App.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ function App() {
     <>
       <Route path="/profile" element={<ProfilePage />} />
       <Route path="/courses" element={<CourseIndexPage />} />
-      <Route path="/schools" element={<SchoolIndexPage />} />
+      <Route path="/courses/create" element={<CoursesCreatePage />} />
     </>
```
Comment (pconrad) [inline frontend/src/App.js]: It looks llike you are removing the schools index and replacing it with /courses/create.

Are you really sure that's what you want to do here?

File: frontend/src/main/components/Courses/CoursesForm.js [hunk 2]
```diff
@@ -2,13 +2,29 @@ import { Button, Form, Row, Col } from 'react-bootstrap';
 import { useForm } from 'react-hook-form'
 import { useNavigate } from 'react-router-dom'
 import { enableEndDateValidation } from './dateValidation'; // Import the JavaScript file
-import React, { useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 
-function CoursesForm({ initialContents, submitAction, buttonLabel = "Create" }) {
+// import { schoolsFixtures } from "fixtures/schoolsFixtures.js";
```
Comment (pconrad) [inline frontend/src/main/components/Courses/CoursesForm.js]: Commented out code

File: frontend/src/tests/components/Courses/CoursesForm.test.js [hunk 3]
```diff
@@ -122,4 +145,45 @@ describe("CoursesForm tests", () => {
 
     });
 
+    //to be added for mutation
+    //line 24 in coursesform.js: [] changed to ["Stryker was here"]
+    // Added this test to address the mutation testing issue
```
Comment (pconrad) [inline frontend/src/tests/components/Courses/CoursesForm.test.js]: I don't think this comment really adds any value for future developers.

This is the kind of thing you can comment on in the PR if you want to document it for folks reviewing your code. 

In particular: line numbers are going to change, so as my former boss at AppFolio likes to say, this comment is a "lie waiting to happen".

**Review by pconrad — CHANGES_REQUESTED** [review]
I'm having trouble testing this since the dokku deployment doesn't seem to have a place for creating courses.

I'm also confused about "selecting classes" in the title.  Where is there a use case in this app for  selecting classes?  Do you mean schools.

Several more things to address:
* You seem to be removing a route for the SchoolsIndexPage.  Are you sure you want to do that?
* Some comments that don't seem to add value.  See the details in the code review.
* Finally: please read: https://ucsb-cs156.github.io/topics/pull_requests/package_lock_json.html and make the necessary adjustments to this PR.

**Review by JUstISAGE — DISMISSED** [review]
LGTM

**Comment by ylin250** [comment]
after looking through dokku logs, We reached the conclusion that it's likely that deployed postgres database doesn't have auto increment capability

**Comment by ylin250** [comment]
PR description has been fixed, please take a look

**Comment by pconrad** [comment]
Replaced by #39

**Pull request closed without merging**